src/ActionButton/ActionButton.js

```diff
@@ -10,47 +11,109 @@ export default class WrappedActionButton extends Component {
     color: '#fff',
   }
 
-  renderSub() {
-    let { color, backgroundColor, icon, action } = this.props
+  state = {
+    width: null,
+  }
+
+  handleLayout = ({
+    nativeEvent: {
+      layout: { width },
+    },
+  }) => {
+    if (!this.state.width !== width) {
+      this.setState({ width })
+    }
+  }
+
+  renderPaper() {
+    let {
+      text,
+      icon,
+      color,
+      backgroundColor,
+      action,
+      _fonts,
+      editor,
+      _width,
+      buttonType,
+      resizeMethod,
+    } = this.props
+    let { width } = this.state
 
     let containerStyles = {
       backgroundColor,
-      position: 'relative',
-      bottom: 'auto',
-      right: 'auto',
+      height: 56,
+      flexDirection: 'row',
+      alignSelf: 'stretch',
+      alignItems: 'center',
+      justifyContent: 'stretch',
     }
 
-    let iconStyles = { color }
+    let offset = 0
+
+    if (!editor && buttonType === 'extended' && width) {
+      switch (resizeMethod) {
+        case 'left':
+          offset = width - _width
+          break
+        case 'right':
+          offset = 0
+          break
+        case 'center':
+          offset = (width - _width) / 2
+      }
+    }
+
+    let wrapperStyles = {
+      alignItems: 'center',
+      justifyContent: 'stretch',
+      flexDirection: 'row',
+      height: 56,
+      minWidth: 56,
+      right: offset,
+    }
+
+    if (Platform.OS === 'ios' || Platform.OS === 'android') {
+      delete containerStyles.justifyContent
+      delete wrapperStyles.justifyContent
+    }
+
+    let labelStyles = {}
+    if (this.props.styles && this.props.styles.text) {
+      labelStyles = {
+        ...this.props.styles.text,
+      }
+    } else if (_fonts) {
+      labelStyles = {
+        fontFamily: _fonts.body,
+      }
+    }
+
+    let breakless =
```

Comment:
Can we turn `let`s to `const`s? I think `breakless`, `wrapperStyles`, `containerStyles`, and the destructured `prop`s can be `const` bindings, maybe other things as well.

Reply:
Yeah I can clean this up